R/prep_data.R

```diff
@@ -41,12 +41,15 @@
 #'   0.0167. It would be excluded by default or if
 #'   `remove_near_zero_variance` > 0.0166. Larger values will remove more columns
 #'   and this value must lie between 0 and 1.
-#' @param convert_dates Logical or character. If TRUE (default), date columns
-#'   are identifed and used to generate day-of-week, month, and year columns,
-#'   and the original date columns are removed. If FALSE, date columns are
-#'   removed. If a character vector, it is passed to the `features` argument of
-#'   `recipes::step_date`. E.g. if you want only quarter and year back:
-#'   `convert_dates = c("quarter", "year")`.
+#' @param convert_dates character. It specifies how `prep_data` handles date and
```

Comment:
@mmastand, I am trying hard to use these the right ways. Can you verify that I know when to use these?

* \code{} - can only be used in documentation (not the example comments). It is used to refer to other functions only.
* single quotes - usually I use this ``, not ''. Is this `` what you are talking about when you say single quotes? I have been using these for anything that is code.
* double quotes - I think I am suppose to use these for values of variables. 

If this was in documentation, is this correct? (just to illustrate, not a phrase I would use)
`convert_dates` is a parameter of  \code{prep_data} that can be set to "continuous", "categories", or "none".

Reply:
Yes, you have it right in that last sentence. The backtick, ` (same button as ~), has a special function in markdown and can be confusing to use it as a quote. Roxygen doesn't format it so it looks like you meant to use \code{} but screwed up. Notice how above, __convert_dates__ got formatted by markdown, but __\code{prep_data}__ did not. It would be opposite in Roxygen. I think what you mean to do is have both things formatted as code, in which case, you'll need a __\code{}__ in both places.

If you want to quote a variable name, use double quotes as you have. When I say single quotes, I mean the apostrophe. In R, the follow is equivalent:

myName <- 'mike'
myName <- "mike"

That can be confusing and the tidyverse style is to always use double quotes. Probably to distinguish from backticks, or maybe SQL code, which uses single quotes.

Reply:
Thanks for taking the time to explain this! So to clarify, in the Roxygen I will hardly ever use the back ticks. This is because it will not format it. Instead I will be using \code{}. I reworked this to say what I wanted to say better, so I won't be able to show you that I learned what you said. I will implement this in the future.